meson.build

```diff
@@ -0,0 +1,121 @@
+# Meson build file
+
+# https://github.com/linuxmint/cinnamon-desktop
+project('cinnamon-desktop', 'c', version : '3.5.0',
+  meson_version : '>=0.40.0'
```

Comment:
Can the required meson version be reduced to accommodate LMDE3

Reply:
I tried to use 0.37.0, but 0.40.0 provides `link_whole`.
http://mesonbuild.com/Release-notes-for-0-40-0.html

Reply:
Can you replace link_whole?, this really needs to build on lmde3

Reply:
Not really, no. I even waited for Meson to implement this feature.
That's compiler-level, it passes `-Wl,--whole-archive` to the linker. It's needed to expose archive symbols in the shared libraries dynamic symbol table.

By the way, Meson is a really young project with really fast changes. 0.40 was an important release, you can't really *not* ship it with LMDE3.

Reply:
Most of the Meson projects will target `0.40.x`. Here : 
* gstreamer https://cgit.freedesktop.org/gstreamer/gstreamer/tree/meson.build
* systemd https://github.com/keszybz/systemd/blob/c4f082a7f87ba888a28525b43921754e1bf29413/meson.build